projects/cdk/directives/auto-focus/autofocus.options.ts

```diff
@@ -22,36 +22,32 @@ export const TUI_AUTOFOCUS_HANDLER = new InjectionToken<TuiAutofocusHandler>(
     'Autofocusing handler',
 );
 
-// TODO: 3.0 remove in ivy compilation
-export function tuiAutofocusHandlerFactory(
-    tuiFocusableComponent: TuiFocusableElementAccessor | null,
-    elementRef: ElementRef<HTMLElement>,
-    animationFrame$: Observable<number>,
-    renderer: Renderer2,
-    ngZone: NgZone,
-    windowRef: Window,
-    isIos: boolean,
-): TuiAutofocusHandler {
-    return isIos
-        ? new TuiIosAutofocusHandler(
-              tuiFocusableComponent,
-              elementRef,
-              renderer,
-              ngZone,
-              windowRef,
-          )
-        : new TuiDefaultAutofocusHandler(
-              tuiFocusableComponent,
-              elementRef,
-              animationFrame$,
-          );
-}
-
 export const TUI_AUTOFOCUS_PROVIDERS = [
     {
         provide: TUI_AUTOFOCUS_HANDLER,
         // TODO: replace to useClass in v3.0
-        useFactory: tuiAutofocusHandlerFactory,
+        useFactory: (
+            tuiFocusableComponent: TuiFocusableElementAccessor | null,
+            elementRef: ElementRef<HTMLElement>,
+            animationFrame$: Observable<number>,
+            renderer: Renderer2,
+            ngZone: NgZone,
+            windowRef: Window,
+            isIos: boolean,
+        ) =>
+            isIos
+                ? new TuiIosAutofocusHandler(
```

Comment:
@splincode were we waiting for Ivy to switch to useClass here?

Reply:
yes)

Reply:
I saw todo but I didn't catch that, guys
what class shout I use to replace it? what problem there was before Ivy?

Reply:
it will be done in next pr